app/meteor-accounts-saml/server/lib/Utils.ts

```diff
@@ -451,7 +451,12 @@ export class SAMLUtils {
 		}
 
 		if (roleAttributeName && profile[roleAttributeName]) {
-			userObject.roles = this.ensureArray<string>((profile[roleAttributeName] || '').split(','));
+			let value = profile[roleAttributeName] || '';
```

Comment:
Splitting an empty string results in `[""]`. Maybe use `[]` instead of `''` as a default since you are now checking its type? That way you don't get a weird empty role.